test/functional/http/get/dapps.js

```diff
@@ -1,460 +1,488 @@
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('dapps').that.is.an('array');
+					node.expect(res.dapps).to.have.length.at.most(100);
+				});
 			});
 		});
-	});
 
-	describe('?limit=1&offset=', function () {
+		describe('limit=1&', function () {
 
-		function getDapps (params, done) {
-			http.get('/api/dapps?limit=1&offset=' + params, done);
-		}
+			it('using offset < 0 should fail', function () {
+				var params = [
+					'limit=' + 1,
+					'offset=' + '-1'
+				];
 
-		it('using offset < 0 should fail', function (done) {
-			var offset = -1;
-
-			getDapps(offset, function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.not.ok;
-				node.expect(res.body).to.have.property('error').that.is.equal('Value -1 is less than minimum 0: #/offset');
-				done();
+				return getDappsPromise(params).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('error').that.is.equal('Value -1 is less than minimum 0: #/offset');
+				});
 			});
-		});
 
-		it('using offset == 0 should be ok', function (done) {
-			var offset = 0;
+			it('using offset 0 should be ok', function () {
+				var params = [
+					'limit=' + 1,
+					'offset=' + 0
+				];
 
-			getDapps(offset, function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.ok;
-				node.expect(res.body).to.have.property('dapps').that.is.an('array').and.has.lengthOf(1);
-				done();
+				return getDappsPromise(params).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('dapps').that.is.an('array').and.has.lengthOf(1);
+				});
 			});
-		});
 
-		it('using offset == 1 should be ok', function (done) {
-			var offset = 1;
+			it('using offset 1 should be ok', function () {
+				var params = [
+					'limit=' + 1,
+					'offset=' + 1
+				];
 
-			getDapps(offset, function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.ok;
-				node.expect(res.body).to.have.property('dapps').that.is.an('array').and.has.lengthOf(1);
-				done();
+				return getDappsPromise(params).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('dapps').that.is.an('array').and.has.lengthOf(1);
+				});
 			});
 		});
-	});
 
-	describe('?orderBy=', function () {
+		describe('orderBy=', function () {
 
-		function getDapps (params, done) {
-			http.get('/api/dapps?orderBy=' + params, done);
-		}
+			it('using "category:asc" should be ok', function () {
+				var params = [
+					'category:asc'
+				];
+				
+				return getDappsPromise(params).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('dapps').that.is.an('array');
+					node.expect(res.dapps[0].category).to.be.at.least(res.dapps[1].category);
```

Comment:
Why are we checking category with .to.be.at.least assertion?

Reply:
Because it can also be equal.

Reply:
i mean we are checking whether sorting works or not. And only checking if first two elements are in correct order. That's not right.